### ISSUE TITLE
Repl doesn't work properly

### ISSUE THREAD

**Issue by Menschenkindlein** [minimal-xyz/minimal-shadow-cljs-webpack#7]
A lot of thanks for this example! It helps a lot.

There was only one problem with `shadow-cljs.edn` - `:entries` line is missing, but from the error message it's easy enough to understand what's going wrong and fix it.

I'd like to ask for help with another issue: I add required dependencies for repl:

```edn
                [nrepl "0.6.0"]
                [cider/cider-nrepl "0.25.0-alpha1"]
```

After that I can connect to the repl and even can evaluate some expressions. But for anything more complex than `(+ 1 2)` or `(js/console.log "hello world")`, there is an error in browser console:

```
failed to parse websocket message {:type :repl/invoke, :name "<eval>", :js "cider.nrepl.....a lot of text here......", :warnings [], :id 10} ReferenceError: CLOSURE_BASE_PATH is not defined
```

Is it a problem in interaction between webpack and shadow-cljs?

**Comment by jiyinyiyong** [comment]
I'm not a Cider user sorry I can't offer help. You'd better add a reproducable demo and report to https://github.com/thheller/shadow-cljs . Since Webpack and shadow-cljs are two standalone tools I'm not sure if they interact with each other.

> There was only one problem with shadow-cljs.edn - :entries line is missing, but from the error message it's easy enough to understand what's going wrong and fix it.

I'm not sure what did you mean. `:entries` is used when shadow-cljs is running a bundled cljs script in a browser, but in this demo shadow-cljs just compiles all code to js and I don't see entries. What error did you encounter?

**Comment by Menschenkindlein** [comment]
It fails the same way with `shadow-cljs cljs-repl app`.

**Comment by Menschenkindlein** [comment]
Regarding `:entries`:

```
$ npm run watch

> minimal-cljs-webpack-hmr@0.1.0 watch ~/minimal-shadow-cljs-webpack
> shadow-cljs watch app

shadow-cljs - config: ~/minimal-shadow-cljs-webpack/shadow-cljs.edn
shadow-cljs - server version: 2.9.10 running at http://localhost:9630
shadow-cljs - nREPL server started on port 49555
shadow-cljs - watching build :app
[:app] Configuring build.
[:app] Build failure:
Invalid configuration
-- Spec failed --------------------

  {:target :npm-module, :output-dir "compiled/", :build-id :app}

should contain key: :entries

| key      | spec            |
|==========+=================|
| :entries | (coll-of        |
|          |  simple-symbol? |
|          |  :distinct      |
|          |  true           |
|          |  :min-count     |
|          |  1              |
|          |  :kind          |
|          |  vector?)       |

-- Relevant specs -------

:shadow.build.targets.npm-module/target:
  (clojure.spec.alpha/keys
   :req-un
   [:shadow.build.targets.shared/output-dir
    :shadow.build.targets.npm-module/entries]
   :opt-un
   [:shadow.build.targets.npm-module/runtime
    :shadow.build.targets.shared/devtools])
:shadow.build.config/build+target:
  (clojure.spec.alpha/and
   :shadow.build.config/build
   (clojure.spec.alpha/multi-spec
    shadow.build.config/target-spec
    :target))

-------------------------
Detected 1 error
```

**Comment by jiyinyiyong** [comment]
I think it's because you are using npm and the project was using `yarn.lock` specifying shadow-cljs `2.8.x` is used. In `2.9.x` things change.

* https://github.com/thheller/shadow-cljs/commit/2725d0a11a19b081b5e2477576c42db28792b753
* https://github.com/thheller/shadow-cljs/releases/tag/2.9.0

**Comment by Menschenkindlein** [comment]
The issue was resolved after upgrading to shadow-cljs 2.10.6
